### DISCUSSION TITLE
[質問]: リストのプラスボタンを押すとクラッシュする

### DISCUSSION THREAD

**Discussion by sheep1857** [lit-iphone-course/lit-iphone-course#154] [Question-original]
### エラーの概要

シュミレーターで実行し、リストのプラスボタンを押すと写真のようなエラーが出てクラッシュしてしまいます。

<img width="870" alt="スクリーンショット 2023-04-05 18 37 49" src="https://user-images.githubusercontent.com/105029351/230043673-82666119-518e-46f9-a29b-c5cd775f3abb.png">

### GitHubのURL

https://github.com/sheep1857/GoshuinApp.git

### TODOリスト

- [X] GitHubに最新のコードをアップロードしました。
- [X] できるだけ具体的なタイトルを設定しました。

**Comment by txxxxc** [comment]
(深夜の返信ごめんなさい><)
手元でエラーが再現できなかったので予想になってしまうんですが、多分Realmのデータ構造を変更したことが原因だと思うので、一度シミュレータからアプリを消してもう一回実行してもらうのを試してみてほしいです！！

![image](https://user-images.githubusercontent.com/22112440/230142919-11944bf7-f1c9-41d8-9f35-f49372c4ff30.png)


コンソールとかにもう少しエラーの詳細が出てるはずなので、共有してもらえると助かります...!

**Comment by txxxxc** [comment]
こちら解決できそうでしょうか...?
ヤバそうだったら教えて下さい！！

**Reply by sheep1857** [reply]
<img width="673" alt="スクリーンショット 2023-04-07 12 58 48" src="https://user-images.githubusercontent.com/105029351/230538299-18cd9759-503f-4b8c-9ab3-385aefdb1a0e.png">

一度アプリを削除してみたのですが、やはり同じところでクラッシュしてしまいます

**Reply by txxxxc** [reply]
見た感じclassの関連付けで失敗してそうなので、もう一回関連付け確認してみてくださいー！！

**Reply by sheep1857** [reply]
ListTableViewの関連付けが間違っていたので直したところ、ホームからListTableViewに飛べなくなってしまいました、、

**Reply by txxxxc** [reply]
なるほど！！最新のコードGitHubにあげてもらえると助かりますー！

**Reply by sheep1857** [reply]
https://github.com/sheep1857/GoshuinApp.git
こちらプッシュしました！

**Reply by txxxxc** [reply]
返信遅れました><(また深夜でごめんなさい🙇‍♂️)
おそらく、UITableViewControllerを継承しているListTableViewControllerとStoryboard上でUIViewControllerを関連付けしている所が原因だと考えられます！
UITableViewControllerとUIViewControllerは別物なので、どちらかに揃えてあげてください！！
<img width="931" alt="image" src="https://user-images.githubusercontent.com/22112440/230633708-724e3f78-b7cc-494b-a96d-4bd5aec5e9ad.png">

**Reply by txxxxc** [reply]
<img width="789" alt="image" src="https://user-images.githubusercontent.com/22112440/230634096-39c0b3ee-7ae0-4f25-9c63-49c46569f550.png">

**Reply by sheep1857** [reply]
https://github.com/sheep1857/GoshuinApp.git

UITableViewControllerで作り直したのですが、リストから探すボタンを押すとクラッシュしてしまいます、、

**Reply by txxxxc** [reply]
(毎回深夜でごめんなさい><)
RealmDataクラスのinit関数が原因っぽいので、以下の画像のようにコメントアウトしてみてくださいー！！
![image](https://user-images.githubusercontent.com/22112440/230731112-e2a07659-5cb7-405c-8e36-f77e6e38ed0b.png)

**Reply by sheep1857** [reply]
クラッシュせずに実行できました。ありがとうございます
